src/matlab2tikz.m

```diff
@@ -922,6 +922,14 @@ case guitypes()
     m2t = drawBoxAndLineLocationsOfAxes(m2t, handle);
     m2t = drawGridOfAxes(m2t, handle);
     m2t = drawLegendOptionsOfAxes(m2t);
+    
```

Comment:
I think this should be moved above to the following clause

```matlab
if ~isVisible(handle)
```
and an elseif. I would prefer if hide axis is not accompanied with 'axis xyz line = 0'